python/sparkdl/graph_builder.py

```diff
@@ -0,0 +1,329 @@
+#
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import joblib as jl
+import logging
+from pathlib import Path
+import shutil
+import six
+from tempfile import mkdtemp
+import webbrowser
+
+import keras.backend as K
+import tensorflow as tf
+import tensorframes.core as tfrm
+
+from .utils import JVMAPI
+
+logger = logging.getLogger('sparkdl')
+
+class GraphBuilder(object):
+    """ 
+    Building TensorFlow graph and export as either UDF or GraphFunction
+    
+    This is a thin layer on top of `tf.Session`.
+    """
+
+    def __init__(self, graph):
+        assert isinstance(graph, tf.Graph)
+        self.sess = tf.Session(graph=graph)
```

Comment:
I combined the `tf.Session` like features of `GraphBuilder` and `GraphBuilderSession` into a single class named `IsolatedSession`.